File: app/src/main/java/com/katkov/lolachievements/data/cloud/api/ApiService.java
package com.katkov.lolachievements.data.cloud.api;

import com.katkov.lolachievements.data.cloud.model.ChampionMasteryDTOApiModel;
import com.katkov.lolachievements.data.cloud.model.SummonerDTOApiModel;
import com.katkov.lolachievements.data.cloud.utils.ApiUtils;

import java.util.List;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface ApiService {

    @GET(ApiUtils.BASE_URL + "summoner/v4/summoners/by-name/{summoner_name}")
    Single<SummonerDTOApiModel> getSummonerDTO(
            @Path("summoner_name") String summonerName,
            @Query("api_key") String apiKey
    );

    @GET(ApiUtils.BASE_URL + "champion-mastery/v4/champion-masteries/by-summoner/{encryptedSummonerId}")
    Single<List<ChampionMasteryDTOApiModel>> getChampionsMasteryDTO(
            @Path("encryptedSummonerId") String encryptedSummonerId,
            @Query("api_key") String apiKey
    );
}

File: app/src/main/java/com/katkov/lolachievements/data/cloud/api/ApiService.kt
package com.katkov.lolachievements.data.cloud.api

import com.katkov.lolachievements.data.cloud.model.ChampionApiModel
import com.katkov.lolachievements.data.cloud.model.MatchlistApiModel
import com.katkov.lolachievements.data.cloud.model.SummonerApiModel
import com.katkov.lolachievements.data.cloud.utils.ApiUtils
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface ApiService {

    @GET(ApiUtils.BASE_URL + "summoner/v4/summoners/by-name/{summoner_name}")
    fun getSummonerApiDto(
        @Path("summoner_name") summonerName: String,
        @Query("api_key") apiKey: String
    ): Single<SummonerApiModel>

    @GET(ApiUtils.BASE_URL + "champion-mastery/v4/champion-masteries/by-summoner/{encryptedSummonerId}")
    fun getChampionApiDto(
        @Path("encryptedSummonerId") encryptedSummonerId: String,
        @Query("api_key") apiKey: String
    ): Single<List<ChampionApiModel>>

    @GET(ApiUtils.BASE_URL + "match/v4/matchlists/by-account/{encryptedAccountId}")
    fun getMatchListApiDto(
        @Path("encryptedAccountId") encryptedAccountId: String,
        @Query("api_key") apiKey: String,
        @Query("beginIndex") beginIndex: Int,
        @Query("endIndex") endIndex: Int
    ): Single<MatchlistApiModel>
}
